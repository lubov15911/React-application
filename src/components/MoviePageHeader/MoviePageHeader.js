import React from 'react';
import PropTypes from 'prop-types';

import HomePageLogoButton from '../HomePageLogoButton/HomePageLogoButton';
import HomePageSearchButton from '../HomePageSearchButton/HomePageSearchButton';
import MovieCard from '../MovieCard/MovieCard';

import './MoviePageHeader.scss';

const MoviePageHeader = (props) => {
    const { routeToHomePage } = props;

    return (
        <div className="movie-container ">
            <div className="buttons-container">
                <HomePageLogoButton handleClick={routeToHomePage} />
                <HomePageSearchButton handleClick={routeToHomePage} />
            </div>
            <MovieCard />
        </div>
    );
};
MoviePageHeader.propTypes = {
    routeToHomePage: PropTypes.func.isRequired,
};

export default MoviePageHeader;
