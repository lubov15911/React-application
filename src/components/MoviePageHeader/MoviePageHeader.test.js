import React from 'react';
import { shallow } from 'enzyme';

import MoviePageHeader from './MoviePageHeader';

describe('MoviePageHeader', () => {
    it('should render correctly', () => {
        const movieId = 354912;
        const fakeFunction = () => {};

        const component = shallow(<MoviePageHeader
            movieId={movieId}
            routeToHomePage={fakeFunction}
        />);

        expect(component).toMatchSnapshot();
    });
});
