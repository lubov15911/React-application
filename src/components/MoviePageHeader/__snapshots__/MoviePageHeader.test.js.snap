// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePageHeader should render correctly with props 1`] = `
<div
  className="movie-container "
>
  <div
    className="buttons-container"
  >
    <HomePageLogoButton
      handleClick={[Function]}
    />
    <HomePageSearchButton
      handleClick={[Function]}
    />
  </div>
  <MovieCard />
</div>
`;
